### ISSUE TITLE
Battery drain in shutdown...?

### ISSUE THREAD

**Issue by horvathcsabalaszlo** [linux-surface/linux-surface#1227]
[Description of the bug or feature]

Anyone else experiencing battery drain while the device is off? As per Google this is a common thing on Windows too (and as i seen, there was no real solution), but i want to as if the kernel developers found any root cause or solution. 
The device got a replacement battery, and it drains in 1-2 days even when in shutdown. Don't know what was the situaton before, as i got it used.

Thanks for any info :)

- Hardware model: Surface Pro 4
- Kernel version: Latest
- Distribution: Debian

**Comment by KTibow** [comment]
Something that worked on my SP7:
Do connected USB devices have power when it's supposed to be shut down? In that case try holding down the power button until they no longer do.

**Comment by horvathcsabalaszlo** [comment]
Hi,


Thanks! I never checked this :) Will come back after i tried.


And, i bet the camera and maybe some more stuff is also hanging on USB in the Surface.

**Comment by horvathcsabalaszlo** [comment]
...tested. USB power ia cut when the device shuts down. Despite this, it drains the battery :(

**Comment by DasSkelett** [comment]
I notice this also since a few weeks, in my case I'm triple booting Windows, Kubuntu 22.04 and Archlinux, on my Surface Go (1).
After cleanly shutting down the device, it shows an empty battery when trying to power it on again after two or three days.
Two things I remember changing in the last few weeks is inserting an SD card and installing Archlinux as third OS.

**Comment by horvathcsabalaszlo** [comment]
I only use Debian (and before Kubuntu) , but i noticed this since the beginning.

**Comment by CoolestEnoch** [comment]
I also fount this problem with my Surface Go 3 running with `Arch Linux`. After I run `powroff` in console, the device SEEMS to be shut down, but actually it DOESN'T. It keeps screen off, keyboard off even USB off. I always power off my surface in this way at night and got it to work at the second day's morning, but I found the location where CPU at is pretty hot. I tried to power it up again with a short hold on the power button, it didn't powered up. But if I hold the power button for a while, it starts.

It seems like not the problem with `linux-surface` kernel, When I was using `linux-zen` kernel, it also appears to be like this.

**How do I do for this**
I always run `reboot` in console, then I power it off when the device booted into `rEFInd`(I use `rEFInd` to boot `grub` and then boot `linux`). If the device turned off the screen but doesn't boot again, I will try the [`SysRq key`](https://wiki.archlinux.org/title/Keyboard_shortcuts) to force it reboot then turn it off in `rEFInd`.

**Comment by python-destroyer** [comment]
I have the same issue on Surface GO (1), Ubuntu 23.10, kernel 6.5.7-surface-2. The problem looks intermittent, as sometimes the battery drains in a shut down state and sometimes not. There is no power on the USB when shut down.

Is there any known solution?

**Comment by ddurdle** [comment]
I have 3 Surface tablets.  2 are Surface GO 1.  On the one that I updated to the latest Debian 12.2  with Linux  6.1.0-13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29) x86_64 GNU/Linux, I've been noticing this 100% of the time when I power off using wayland "power off".  Not happening with an identical Surface GO 1 that is not yet updated to 12.2 (it's running an older Debian 12 release).  I suspect it might be an update recently introduced.  Likewise a Surface Pro 7 running a less than latest Debian release (but is running Surface kernel 6.3) has never had the issue either.

It's very frustrating to be using the device, getting it to 100% in battery, "pwer off" and come back 2 days later in a "pinch" and find you can't turn on the tablet without plugging it in because it's completely drained itself.

All three Surface tablets are running the exact same OS image (I cloned the install to all 3).  The only thing different is the 1 surface tablet I updated to the latest 12.2 back in Sept and I've seen the power drain to 0% the three times I booted the device.

**Comment by ddurdle** [comment]
Checking in from surface go that has the issue:

Linux 6.3.3-surface #1 SMP PREEMPT_DYNAMIC Wed May 17 20:49:28 UTC 2023 x86_64 GNU/Linux
(also issue with kernell -6.1.0-13-amd64)

Debian 12.2 - last updated 10/20

Will chime back in on other devices with their particulars

**Comment by ddurdle** [comment]
Surface go with no power off issue.

Debian 12.0 last updated 04/28
Linux 6.1.0-7-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.20-2 (2023-04-08) x86_64 GNU/Linux

Running no special boot flags.

I'll be doing a full comparison of the bios next.

**Comment by ddurdle** [comment]
I did a comparison between BIOS configs between the two surface tablets.  all the same configuration details (all the same devices enabled/disabled).  Only 1 difference.  The surface go that has the issue is on an older firmware:

bad surface
 system uefi 1.0.32
 intel management engine 11.8.50.3460
system embedded controller 1.0.9.0

good surface
  system uefi 1.0.34
 intel management engine 11.8.90.3987
system embedded controller 1.0.9.0 

I still have the older kernel available on the updated surface, so I'm going to try running it and evaluating the issue.

I can also try to make some space available and load the identical 04/28 install from the good surface on the bad surface to rule in / out debian updates being a culprit.

**Comment by ddurdle** [comment]
Seriously suspecting it is a hardware / firmware issue.  I'm going to try updating the firmware on the suspect surface tablet.  Even if I boot into BIOS and power off from the BIOS, it drains power.

**Comment by oleid** [comment]
> > [Description of the bug or feature]
> > Anyone else experiencing battery drain while the device is off?
> 
> Yes, me!
> 
> > ```
> > * Distribution: Debian
> > ```
> 
> ArchLinux
> 
> I thought the battery was simply broken, however, since I learnt about the drain while off I'm positive that this is the root cause.
> 
> I'm wondering if we're doing something like connected standby instead of powering off. But I'm unsure how to verify.

meanwhile I even wiped my linux installation from the disk, installed re-windows 10 from official media (the HW is too old for win11) and tried certain suggestions I found for windows users - to no avail. I have the drain on windows, too. So I guess there is little one can do, it must be in the firmware.

**Comment by StollD** [comment]
> Sadly there's no option to downgrade UEFI and IME to test older firmwares

There is, but you can't go back all the way because Microsoft marked some older firmwares as unsecure and the newer ones refuse to install them: https://github.com/linux-surface/surface-uefi-firmware

**Comment by RubbelDeCatc** [comment]
Had the same problem with a surface go 3 and the current surface kernel 6.05 ?

An update to the latest firmware seems to help 
`SurfaceGo3_Win11_22000_23.110.4352.0.msi`
found here
[https://www.microsoft.com/en-us/download/details.aspx?id=103504](https://www.microsoft.com/en-us/download/details.aspx?id=103504)

I am unsure. But there is an option to disable secure boot completely now. I can not remember to have this option in my old firmware.

Btw: I used Linux to update the firmware because I wiped windows a long time ago. 
[https://github.com/linux-surface/surface-uefi-firmware](https://github.com/linux-surface/surface-uefi-firmware)
Worked flawless.

**Comment by horvathcsabalaszlo** [comment]
Haven't found any solution. I'm going without SecureBoot, if it counts, but i think it's not. Maybe a battery issue, because i have replaced mine, but I think the others here are having the original.

**Comment by ddurdle** [comment]
Does your surface tablet use EPROM or SSD?  Like I mentioned prior, I have
two surface GO tablets of the same generation.  One has the issue, the
other doesn’t.  I made everything match between units (BIOS updates etc,
BiOS settings).  The only difference between the two units is that one uses
SSD and one uses EPROM.  Starting to think that might me cause.  The unit
that uses EPROM has the battery leak.

**Comment by horvathcsabalaszlo** [comment]
Sry, i don't get that EEPROM vs SSD. Do you referfor eMMC or SSD? 
Pro 4 is using SSD as storage, NVMe m.2 ones. EEPROM is mostly for storing BIOS, so the Pro 4 is using EEPROM for storing BIOS, but almost every computer BIOS is stored in EEPROM :)

**Comment by ddurdle** [comment]
Sorry, yes, I meant my Surface GO unit that has the battery leak issue is
one with eMMC.  That's the only difference between the two units.

**Comment by horvathcsabalaszlo** [comment]
Thanks, then it's clear. It can be a difference, but we need some other cases to be sure.

**Comment by python-destroyer** [comment]
@horvathcsabalaszlo @ddurdle I have a 128GB SSD version with original battery and experience this same issue.

It's really hard to track a correlation with the problem, but I think in my case the active stylus makes some impact. I started taking the battery out of the stylus before I turn off the tablet and the issue now happens less often I think. Do you guys use a stylus?

I suspect this issue is artificially implemented by Microsoft to force users into buying new devices as it's not unique for Surface GO ([Surface Pro](https://answers.microsoft.com/en-us/surface/forum/all/surface-pro-battery-drain-while-off/2820724e-e303-4414-a338-aaa6dcfc329c)), but that's just my wild guess.

**Comment by horvathcsabalaszlo** [comment]
I have a stylus, but it's mostly not in use, and is sitting without battery.

**Comment by RubbelDeCatc** [comment]
> It's really hard to track a correlation with the problem, but I think in my case the active stylus makes some impact. I started taking the battery out of the stylus before I turn off the tablet and the issue now happens less often I think. Do you guys use a stylus?

In my case, it is not the pen and not the battery. I know that there are the same issue with Windows installations. And it is also not related to secure boot. I guess it's a bios or firmware issue in conjunction with the hardware.

If you google for "surface shutdown battery drain" you will find tons of results.

I found this interesting post. There is a part about changing a property of the Intel Management Engine.
https://www.thewindowsclub.com/laptop-battery-drains-after-shutdown-fix

It might be possible to change the behavior of the IME in linux too. Any Idea?

**Comment by horvathcsabalaszlo** [comment]
> > It's really hard to track a correlation with the problem, but I think in my case the active stylus makes some impact. I started taking the battery out of the stylus before I turn off the tablet and the issue now happens less often I think. Do you guys use a stylus?
> 
> In my case, it is not the pen and not the battery. I know that there are the same issue with Windows installations. And it is also not related to secure boot. I guess it's a bios or firmware issue in conjunction with the hardware.
> 
> If you google for "surface shutdown battery drain" you will find tons of results.
> 
> I found this interesting post. There is a part about changing a property of the Intel Management Engine. https://www.thewindowsclub.com/laptop-battery-drains-after-shutdown-fix
> 
> It might be possible to change the behavior of the IME in Linux too. Any idea? Maybe a hack for the DSDT table?

Hm, interesing finding. If someone has Windows also on the computer, can try it. 
I'll check the BIOS, but as i remember, it has no way to access the ME. 
Need to Google about how to insult ME from Linux :)

**Comment by ddurdle** [comment]
The windows IME setting seems probably equivalent of something you can find
via powertop.

Something about the device keeps power flowing even when fully turned off.
Could be simple some component that is failing or there might be revisions
of the boards that behave correctly.  More apt to think there is a design
flaw and failing component in the devices that are draining.

**Comment by horvathcsabalaszlo** [comment]
But also can be a BIOS setting of ME. 
And yes, it can be a firmware bug which was not present in the previous versions. Will check powertop.

**Comment by horvathcsabalaszlo** [comment]
Checked powertop. It's more complicated that i thought, but identified Intel Corporation Sunrise Point-LP CSME HECI #1 PCI  as Intel Management Engine (or its component), but it was already using a power saving feature. (Was turned to "good" by default.)

But the Marvell wireless card was not switched to "good", so i gine this a chance. If the WWOL or such a feature is active, it can drain power.

**Comment by DantesInferi** [comment]
I have Surface Pro 5 running Ubuntu and latest Linux-Surface. It drains itself over 2 days too now. 
It most certainly didn't do it on Windows 11 for the last 3 years - though many users reported that in earlier models on whatever system.

**Comment by AdebayoIbrahim** [comment]
I have a HP PAvilion 15 laptop , when i was using windows nothing like this happened, After i wiped windows and stick to ubuntu , my battery drains after shutdown , I've tried everything including tlp and tlpui..., Still Experiencing this issue though

**Comment by doughdemon** [comment]
Experiencing the same as in https://github.com/linux-surface/linux-surface/issues/1227#issuecomment-1809752223 on my Surface Book 2. Previously reported in #693.

**Comment by Arsenitim** [comment]
Same here. Surface Go 1.
Roughly in 1 of 5 shutdowns the battery is fully depleted overnight

**Comment by theflyingrahul** [comment]
Same issue here. SGO 1 on 6.15.1-surface-2, ubuntu 25.04.
However, the problem was present on stock kernel as well, before I switched to linux-surface.

**Comment by horvathcsabalaszlo** [comment]
Seems to be a HW issue (battery). I replaced my battery, as it inflated while the device wasn't used. The new battery is not affected by this issue. It drains significantly while in standby, but doesn't while in shutdown.

**Comment by surfacego3** [comment]
**NO BATTERY DRAIN.**

Surface Go 3 64GB Edition 4 GB RAM
Fedora KDE Plasma Mobile Spin (only installed os, no dual boot)
Stock Fedora kernel
Linux fedora 6.16.3-200.fc42.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Aug 23 17:02:17 UTC 2025 x86_64 GNU/Linux
Screen scaling 145%

Windows 11, no battery drain.

Previous setup with Ubuntu 24.04
Battery empty over night after shutdown.



Battery drain Kernel parameter issue?

**Comment by horvathcsabalaszlo** [comment]
> **NO BATTERY DRAIN.**
> 
> Surface Go 3 64GB Edition 4 GB RAM Fedora KDE Plasma Mobile Spin (only installed os, no dual boot) Stock Fedora kernel Linux fedora 6.16.3-200.fc42.x86_64 [#1](https://github.com/linux-surface/linux-surface/pull/1) SMP PREEMPT_DYNAMIC Sat Aug 23 17:02:17 UTC 2025 x86_64 GNU/Linux Screen scaling 145%
> 
> Windows 11, no battery drain.
> 
> Previous setup with Ubuntu 24.04 Battery empty over night after shutdown.
> 
> Battery drain Kernel parameter issue?

What happens if you shut down the device from Ubuntu and then power on, go to UEFI and power off from there?

**Comment by theflyingrahul** [comment]
> Seems to be a HW issue (battery). I replaced my battery, as it inflated while the device wasn't used. The new battery is not affected by this issue. It drains significantly while in standby, but doesn't while in shutdown.

I don't think so. Battery is fairly healthy. The issue is absent when I force power down using the power button. Also doesn't exist when I power off from UEFI. Went back to Windows to check, the issue is absent. Looks like a linux/kernel problem. Funnily, the drain is significantly lower on standby for me than shutdown!

**Comment by horvathcsabalaszlo** [comment]
> > Seems to be a HW issue (battery). I replaced my battery, as it inflated while the device wasn't used. The new battery is not affected by this issue. It drains significantly while in standby, but doesn't while in shutdown.
> 
> I don't think so. Battery is fairly healthy. The issue is absent when I force power down using the power button. Also doesn't exist when I power off from UEFI. Went back to Windows to check, the issue is absent. Looks like a linux/kernel problem. Funnily, the drain is significantly lower on standby for me than shutdown!

Hi,
It's strange. I had that battery for 2 years, and it lasted long when the tablet was running. After i replaced, it was OK in standby, only a few % drain overnight. However, the new battery lasted for 3 weeks, and it died now :D 
So in my case it was a battery issue. I wonder how it can be a kernel issue, but seems to be in your case, it is.

**Comment by surfacego3** [comment]
> What happens if you shut down the device from Ubuntu and then power on, go to UEFI and power off from there?

Sorry I got rid of Ubuntu, so do not know.
Fedora KDE Plasma Mobile Spin is optimized for tablets and other small handheld devices, may be the reason it works good also on Surface. Anyone else tried Fedora?

**Comment by horvathcsabalaszlo** [comment]
No, i haven't. But i also had no problem with Ubuntu with the second battery.

**Comment by Rsge** [comment]
Just chiming in: I’ve had _two_ SP4s now, both run with with _pure_ Windows and _pure_ Linux, each for a while, and both, _no matter the OS/boot loader_, had strong battery drain when shut down _from the beginning_.\
So, at least for my case, something related to linux-surface seems unlikely to impossible…\
I'll have to see if a firmware update is possible, but given the age of the devices, it probably isn’t…

**Comment by akavkl** [comment]
The only way I see is to reboot to UEFI then hold power to shutdown.
Works for surface go 1 and surface pro 7.

**Comment by AdebayoIbrahim** [comment]
> The only way I see is to reboot to UEFI then hold power to shutdown.
> Works for surface go 1 and surface pro 7.

Shutting down that way kills the Ssd

**Comment by horvathcsabalaszlo** [comment]
> Just chiming in: I’ve had _two_ SP4s now, both run with with _pure_ Windows and _pure_ Linux, each for a while, and both, _no matter the OS/boot loader_, had strong battery drain when shut down _from the beginning_. So, at least for my case, something related to linux-surface seems unlikely to impossible… I'll have to see if a firmware update is possible, but given the age of the devices, it probably isn’t…

Bad battery :S

**Comment by Rsge** [comment]
> Bad battery :S

Battery time was (and still is) fine when it’s being used after being charged, over 180 mins screen time, Windows and Linux, it just _also_ discharges rather fast (maybe around 24 h?) after explicitly being shut down, which is what this issue is all about I thought…

**Comment by horvathcsabalaszlo** [comment]
Yes, it can be. But it's not always related... After i replaced my battery, the sleep drain was much less, battery uptime was good, but battery died in 2 weeks :D ...sold my Surface :D
